### ISSUE TITLE
Save forward and back-button history for each tab

### ISSUE THREAD

**Issue by gilch** [sienori/Tab-Session-Manager#835]
**Is your feature request related to a problem? Please describe.**

I have too many tabs open and it's slowing down my computer, even with Auto Tab Discard.

The main reason I keep lots of tabs open instead of just using the bookmark all tabs in a window feature is that tabs contain more information than bookmarks do and this information is _lost_ when I close the tab, even if I have a bookmark for it.

I often read long online articles or online textbooks and don't have the time to finish them before I have to switch tasks. They also have footnotes or hyperlinks with more details about a topic. If I follow those links I could open a new tab, and have an even bigger tabs problem, or I can just use the same window and get back to my spot in the textbook with the back button. The browser also remembers my scroll position and form data. But Tab Session Manager (TSM) doesn't restore all of this.

The scroll position already has an issue: #144. Some books will at least let you link to their headers, but not all of them do. Losing my position in the chapter is frustrating, but I can usually find my spot again with some effort. This is a waste of time compared to just leaving the tab open.

But without my back-button history, I could lose not just my position, but also my chapter, and even the fact that I was reading that particular textbook! If I happen to notice something got lost, I can try to find it in my browser history, but that is organized _wrong_. It shows the date the site was _opened_, but not when it was closed! I might have better luck with a web search, but only if I can remember enough to find it.

I can live without the form data, but this would also be nice to have. I don't trust the browser very much with that, so I don't really fill out more than one form at a time, and if I get interrupted when writing something long, I'll copy it out somewhere else. But the browser usually does remember this, and also keeps form data in previous pages I can access with the back button.

**Describe the solution you'd like**
At minimum, I want Tab Session Manager to save and restore the back/forward button history for each tab.

This would be more usable if TSM could display each tab's back/forward button history in its tree view in the TSM button menu. Each tab would become tree a node like the windows are currently, with the leaves being each page in the back/forward history, maybe with the current position in bold or something.

Ideally, TSM would save and restore the entire state of a tab, including back/forward history, scroll position, and form data, and be able to display this without actually opening the tab.

**Describe alternatives you've considered**
TSM is almost unusable for me without back/forward history. Without this, to save my reading history with TSM, I'd have to always follow links in a new tab and use a lot more windows.

Basically, I'd have to use new tabs for what I used to be able to do with back/forward history, and use entire new windows for what used to be tabs, making my tabs problem explode with even more tabs, slowing down my computer even more. But at least I could save sessions and close the windows with TSM. But to save my current tabs, I'd have to expand their history in a new window for each one before I can safe it. Tedious, but doable.

I could also manually log each page I visit, but this interrupts my flow and doesn't even work that well.

**Comment by sienori** [comment]
Everyone wants to save the history of tabs, but due to the limitations of WebExtensions API, this is not possible... #27
